modules/core/04-channel/keeper/packet_test.go

```diff
@@ -1222,16 +1238,24 @@ func (suite *KeeperTestSuite) TestAcknowledgePacket() {
 			suite.SetupTest() // reset
 
 			path = ibctesting.NewPath(suite.chainA, suite.chainB)
+			ctx := suite.chainA.GetContext()
 
 			tc.malleate()
 
 			packetKey := host.PacketAcknowledgementKey(packet.GetDestPort(), packet.GetDestChannel(), packet.GetSequence())
 			proof, proofHeight := path.EndpointB.QueryProof(packetKey)
 
-			err := suite.chainA.App.GetIBCKeeper().ChannelKeeper.AcknowledgePacket(suite.chainA.GetContext(), channelCap, packet, ack.Acknowledgement(), proof, proofHeight)
+			err := suite.chainA.App.GetIBCKeeper().ChannelKeeper.AcknowledgePacket(ctx, channelCap, packet, ack.Acknowledgement(), proof, proofHeight)
```

Comment:
i think we would prefer to grab the context when the function is getting called generally, so that we avoid using a stale context.

Reply:
hmmm seem like everytime i call `getContext()` it returns a fresh context so the event in the EventManager return an empty array each time. I figure the ctx where i fetch the event need to be the same as the ctx i pass in the `TimeoutExecuted`

Reply:
ah, youre right, bc `malleate` resets the context. got it!